script.js

```diff
@@ -1,4 +1,446 @@
+var cardNumberPerPlayer = []; // player's card number
+var numberofWinsPerPlayer = [];
+var numberOfGamesPerPlayer = [];
+var wagerPointSystem = [];
+var cardsPerRound = [];
+var playerCardsTemp = [];
+var newDeck = [];
+
+var indexOfMaxInArray = function (cardNumberPerPlayer) {
+  if (cardNumberPerPlayer.length == 0) {
+    return -1;
+  }
+
+  var max = cardNumberPerPlayer[0];
+  var maxIndex = 0;
+
+  for (var i = 1; i < cardNumberPerPlayer.length; i++) {
+    if (cardNumberPerPlayer[i] > max) {
+      maxIndex = i;
+      max = cardNumberPerPlayer[i];
+    }
+  }
+
+  return maxIndex;
+};
+var incrementNumberOfWinsOfEachPlayer = function (returnIndexOfHighestNumber) {
+  numberofWinsPerPlayer[returnIndexOfHighestNumber] += 1;
+};
+var countWagerPointSystem = function (returnIndexOfHighestNumber) {
+  wagerPointSystem[returnIndexOfHighestNumber] -= wager;
+};
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
+  if (gameMode == GAME_MODE_ENTER_NUMBER_OF_PLAYERS) {
+    numOfPlayers = Number(input);
+    numberofWinsPerPlayer = Array(numOfPlayers).fill(0);
+    numberOfGamesPerPlayer = Array(numOfPlayers).fill(0);
+    wagerPointSystem = Array(numOfPlayers).fill(100);
+    gameMode = GAME_MODE_ENTER_NAME;
+
+    return `Number of Players: ${numOfPlayers}<br><br>Enter Player ${currentPlayer}'s name!`;
+  } else if (gameMode == GAME_MODE_ENTER_NAME) {
+    var name = input;
+    userName.push(name);
+
+    if (userName.length == numOfPlayers) {
+      var playerIndex = 0;
+      var playerNumber = 1;
+      var output = "";
+      while (playerIndex < userName.length) {
+        output = `${output}Player ${playerNumber}: ${userName[playerIndex]}<br>`;
+        playerIndex += 1;
+        playerNumber += 1;
+      }
+
+      gameMode = GAME_MODE_WAGER;
+
+      return `${output}<br>Enter the preferred wager points!`;
+    }
+
+    var nextPlayer = currentPlayer + 1;
+    currentPlayer = nextPlayer;
+
+    return `Player ${nextPlayer}, enter your name!`;
+  } else if (gameMode == GAME_MODE_WAGER) {
+    wager = input;
+    gameMode = GAME_MODE_DEAL_CARDS;
+    return `Wager: ${wager} points<br><br>Click on "Submit" to deal the cards!`;
+  } else if (gameMode == GAME_MODE_DEAL_CARDS) {
+    cardNumberPerPlayer = [];
+    cardsPerRound = [];
+    currentPlayer = 1;
+    var shuffledDeck = shuffleCards();
+    newDeck = shuffledDeck;
+    var numberOfCardsToBeDealed = numOfPlayers * 2;
+    var cardIndex = 0;
+    while (cardIndex < numberOfCardsToBeDealed) {
+      var card = newDeck.pop();
+
+      if (card.name == "Jack" || card.name == "Queen" || card.name == "King") {
+        card.rank = 10;
+      } else if (card.name == "Ace") {
+        card.rank = " ";
+      }
+
+      cardsPerRound.push(card);
+      cardIndex += 1;
+    }
+    gameMode = GAME_MODE_DECISION;
+    return `All cards have been dealt.<br><br>${
+      userName[currentPlayer - 1]
+    }, click "Submit" to review your cards.`;
```

Comment:
for example, this code here is actually quite logic intensive given the nested loops and if statements. Perhaps this could be its own helper function